middle_end/flambda2/simplify/simplify_apply_expr.ml

```diff
@@ -854,19 +854,21 @@ let rebuild_function_call_where_callee's_type_unavailable apply call_kind
          Unused_because_function_unknown)
   in
   let uacc, expr =
-    EB.rewrite_fixed_arity_apply uacc ~use_id (Apply.return_arity apply) apply
+    match use_id with
+    | Some use_id ->
+      EB.rewrite_fixed_arity_apply uacc ~use_id (Apply.return_arity apply) apply
+    | None ->
+      let uacc =
+        UA.add_free_names uacc (Apply.free_names apply)
+        |> UA.notify_added ~code_size:(Code_size.apply apply)
+      in
+      uacc, RE.create_apply (UA.are_rebuilding_terms uacc) apply
```

Comment:
`Expr_builder.rewrite_fixed_arity_apply` already handles non-returning calls (by doing exactly what this code does), so I'd prefer if we could avoid the code deduplication: why not change `Expr_builder.rewrite_fixed_arity_apply` to take a optional `use_id` as argument ?